src/session.cpp

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include <src/session.hpp>
+
+namespace obmc
+{
+namespace session
+{
+void SessionItem::delete_()
+{
+    auto manager = managerWeakPtr.lock();
+    if (manager)
+    {
+        SessionManager::SessionIdentifier sessionId =
+            SessionManager::parseSessionId(this->sessionID());
+        if (!manager->remove(sessionId)) {
+            throw InternalFailure();
+        }
+    }
+}
+
+void SessionItem::resetCleanupFn(SessionManager::SessionCleanupFn&& cleanup)
+{
+    this->cleanupFn.reset();
+    this->cleanupFn.emplace(cleanup);
+}
+
+void SessionItem::adjustSessionOwner(uid_t)
+{
+    // Define interface to adjust session owner by user ID.
+    // Currently, not implemented.
+    throw std::runtime_error("SessionItem::adjustSessionOwner(uid_t) is not implemented");
+}
+
+void SessionItem::adjustSessionOwner(const std::string& userName)
+{
+    using DBusGetObjectOut = std::map<std::string, std::vector<std::string>>;
+
+    constexpr const std::array userObjectIfaces = {
+        "xyz.openbmc_project.User.Attributes"};
+    auto userObjectPath = "/xyz/openbmc_project/user/" + userName;
+
+    try
+    {
+        DBusGetObjectOut getUserObject;
+        auto callMethod = bus.new_method_call(
+            "xyz.openbmc_project.ObjectMapper",
+            "/xyz/openbmc_project/object_mapper",
+            "xyz.openbmc_project.ObjectMapper", "GetObject");
+        callMethod.append(userObjectPath.c_str(), userObjectIfaces);
+        bus.call(callMethod).read(getUserObject);
+
+        if (getUserObject.empty())
+        {
+            throw std::runtime_error("The username '" + userName +
+                                     "' is not found");
+        }
+
+        this->associations({
+            make_tuple("user", "session", userObjectPath),
+        });
+    }
+    catch (const std::exception& ex)
+    {
+        throw std::runtime_error(ex.what());
```

Comment:
It looks strange. Likely, you  don't need the `try-catch` statement at all here.

Reply:
I've aimed to feel free of sdbusplus API for `libobmcsession` consumers. I don't desire to include sdbusplus headers only to catch exceptions.

Reply:
You are catching it as std::exception and it doesn't require `sdbusplus` API usage.
And one more thing, what() contains the full description of the original exception, including the exception class name. And you are copying it to the new instance. So, I don't see the difference.
It seems to me you can skip this try-catch statement and catch std::exception in the code where you are planing to do it in the same way as you are doing it here.

Reply:
I totally agree with @nest1ing. This looks weird.

Reply:
Fixed